src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java

```diff
@@ -51,11 +52,18 @@
         synchronized (connection.getLock()) {
             connection.getChannel().getCloseSetter().set(new ChannelListener<java.nio.channels.Channel>() {
                 public void handleEvent(final java.nio.channels.Channel channel) {
-                    connection.getExecutor().execute(new Runnable() {
-                        public void run() {
-                            handler.handleConnectionClose();
-                        }
-                    });
+                    try {
+                        connection.getExecutor().execute(new Runnable() {
+                            public void run() {
+                                handler.handleConnectionClose();
+                            }
+                        });
+                    } catch (RejectedExecutionException re) {
+                        log.trace("Conenction already closing.");
+                        // this could only happen if the threadpool is shutting
+                        // down as it us using unbound queue to accept tasks
+                        // the connection must be already closing
+                    }
```

Comment:
This just hides the symptom, it doesn't solve the problem.  If this exception is thrown, then the handler's close method is never actually called, potentially leaving many threads hanging.

The proper fix is to determine how the thread pool is being shut down before the connections are cleaned up.